### ISSUE TITLE
[Bug] Cannot run "php artisan backpack:crud bug" working with Bug.php model

### ISSUE THREAD

**Issue by IazLur** [Laravel-Backpack/Generators#101]
ErrorException 

  Undefined offset: 1

  at C:\wamp64\www\toplitz\vendor\backpack\generators\src\Console\Commands\CrudModelBackpackCommand.php:84
     80▕                 if (Str::contains($line, $classDefinition)) {
     81▕                     if (Str::endsWith($line, '{')) {
     82▕                         // add the trait on the next
     83▕                         $position = $key + 1;
  ➜  84▕                     } elseif ($file_array[$key + 1] == '{') {
     85▕                         // add the trait on the next next line
     86▕                         $position = $key + 2;
     87▕                     }
     88▕

  1   C:\wamp64\www\toplitz\vendor\backpack\generators\src\Console\Commands\CrudModelBackpackCommand.php:84
      Illuminate\Foundation\Bootstrap\HandleExceptions::handleError("Undefined offset: 1", "C:\wamp64\www\toplitz\vendor\backpack\generators\src\Console\Commands\CrudModelBackpackCommand.php")

  2   C:\wamp64\www\toplitz\vendor\laravel\framework\src\Illuminate\Container\BoundMethod.php:37
      Backpack\Generators\Console\Commands\CrudModelBackpackCommand::handle()

**Comment by pxpm** [comment]
Hello @IazLur 

Thanks for the bug report. 

I just transfered your issue to the correct repository. 

Could you please provide some more info with: `php artisan backpack:version` and gives us info what version of generators you are loading in your `composer.json` ?

Thanks in advance,
Pedro

**Comment by IazLur** [comment]
### PHP VERSION:
PHP 7.3.12 (cli) (built: Nov 19 2019 13:58:02) ( ZTS MSVC15 (Visual C++ 2017) x64 )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.12, Copyright (c) 1998-2018 Zend Technologies

### LARAVEL VERSION:
v8.6.0@a71952a6dba55de0bb11b5fbbd84874eda2a755c

### BACKPACK VERSION:
4.1.25@23fb1ed5153d5465d49d2fe1ca265a5985c78877


"require-dev": {
        "backpack/generators": "^3.1",
        "facade/ignition": "^2.3.6",
        "fzaninotto/faker": "^1.9.1",
        "laracasts/generators": "^2.0",
        "mockery/mockery": "^1.3.1",
        "nunomaduro/collision": "^5.0",
        "phpunit/phpunit": "^9.3"
    },

@pxpm

**Comment by tabacitu** [comment]
Hmm that's weird. It worked fine for me:
<img width="603" alt="Screenshot 2020-09-29 at 14 01 29" src="https://user-images.githubusercontent.com/1032474/94598018-56b6cf00-025c-11eb-8b00-ca3a126ecfdd.png">

@IazLur by any chance did you already have a `Bug.php` model? If so, we're going to need the code inside `Bug.php` BEFORE Backpack made any changes to it, to detect what the problem was... What happens is the Backpack tries in this case to add use the `CrudTrait` inside your existing model. But, depending on how your model is coded, it might fail in some very particular cases. We just need to understand what that particular case is 😄

**Comment by IazLur** [comment]
Hi, I tried to understand where was the bug but I didn't found.
Here id the Bug.php before I made the command.

`
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Illuminate\Database\Eloquent\Relations\BelongsTo;;

class Bug extends Model
{
    use HasFactory;

    /**
     * @var string
     */
    protected $table = 'bugs';

    /**
     * @var array
     */
    protected $guarded = [];

    /**
     * @return BelongsTo
     */
    protected function bugGame()
    {
        return $this->belongsTo(BugGame::class);
    }

    /**
     * @return HasMany
     */
    protected function bugMedias()
    {
        return $this->hasMany(BugMedia::class);
    }

    /**
     * @return BelongsTo
     */
    protected function bugType()
    {
        return $this->belongsTo(BugType::class);
    }

    /**
     * @return BelongsTo
     */
    protected function gameVersion()
    {
        return $this->belongsTo(GameVersion::class);
    }
}
`

**Comment by dividy** [comment]
Your model already exists.

Remove it and try again the command.

**Comment by dividy** [comment]
On my side, the bug seems to be with the file being read as 1 big line.

Problem might be here : 
```php
$file_array = explode(PHP_EOL, $file);
```

**Comment by pxpm** [comment]
@dividy could you check if #100 fixes it? 

Best,
Pedro

**Comment by dividy** [comment]
I was about to submit this : 
```php
$file_array = preg_split('/\r\n|\r|\n/', $file);
```

Beat me to it 😄

Your PR works 👍  

> Model already exists! We just added CrudTrait on it.
> Request already exists!
> Route already exists!
> Sidebar item already exists!

**Comment by tabacitu** [comment]
Thanks guys! 

The PR was merged and tagged, a `composer update` should fix it (by getting backpack/generators 3.1.4). So let's close this.

Cheers!
